external/vulkancts/framework/vulkan/vkImageUtil.cpp

```diff
@@ -33,6 +33,7 @@
 
 #include <map>
 #include <assert.h>
+#define VK_SAMPLER_CREATE_NON_SEAMLESS_CUBE_MAP_BIT_EXT 4
```

Comment:
Apologies for this incredibly dirty hack, but I can't figure out which part of the framework and how to regenerate the relevant file where this should be located. Half a year ago I got it to compile somehow but it seems like I'm missing some step because this file now includes `framework/vulkan/generated/vulkansc/vkBasicTypes.inl` instead of `framework/vulkan/generated/vulkan/vkBasicTypes.inl`.

Reply:
Related to this issue, you now need to put the SHA of the vulkan repo that contains the relevant header for this extension under external/fetch_sources.py (sha from this commit for this extension https://github.com/KhronosGroup/Vulkan-Docs/pull/1628). After this you need to regen the inl files with:

python3 scripts/check_build_sanity.py -r gen-inl-files -t <build_dir>

This extension is not testable with Vulkan SC, so it cannot be part of the Vulkan SC build. You'll need to add changes to remove it. I will comment on some places for this.

Reply:
Because of the fetch sources changes, we cannot merge this PR until the vulkan changes are merged first and we have a SHA that is on vulkan-docs/main. For now can use the SHA from the PR for review.

Reply:
Thanks a lot for the help! I implemented all your suggestions regarding VulkanSC, and the branch now compiles without this hack. But I would like to avoid messing around with the submodules unless necessary, is everything okay for review without it?

Reply:
It's likely going to fail the internal CI if the inl files do not match the ones generated from the submodule. My suggestion is to get this building with generated inl files with the PR sha, once the vulkan-docs PR merges it shouldn't cause issues switching to the main SHA.

Reply:
Okay, I hope everything is okay now.